File: server/modules/helpers/service_checker.py
import subprocess

class Service:
    @staticmethod
    def check(service:str) -> bool:
        try:
            result = subprocess.run(
                ["systemctl", "is-active", service],
                stdout=subprocess.PIPE,
                stderr=subprocess.PIPE,
                text=True
            )

            if result.stdout.strip() == "active":
                return True
            else:
                return False
        except Exception as error:
            print("<ERR> [service_checker.py](func check): ", error)
            return False
        
    @staticmethod
    def restart(service: str) -> bool:
        try:
            result = subprocess.run(
                ["systemctl", "restart", service],
                stdout=subprocess.PIPE,
                stdin=subprocess.PIPE,
                text=True
            )

            if result.returncode == 0:
                return True
            else:
                return False
        except Exception as error:
            print("<ERR> [service_checker.py](func restart): ", error)
            return False

File: src/modules/utils/serviceChecker.js
import { spawnSync } from 'child_process';

export default class Service {
    static check(service) {
        try {
            const result = spawnSync('systemctl', ['is-active', service], { encoding: 'utf-8' });
            
            return result.stdout.trim() === 'active';
        } catch(err) {
            console.error(`[ERROR] <Service.check> ${service}: `, err.message);
            return false;
        }
    }

    static restart(service) {
        try {
            const result = spawnSync('systemctl', ['restart', service], { encoding: 'utf-8' });
            return result.status === 0;
        } catch(err) {
            console.error(`[ERROR] <Service.restart> ${service}: `, err.message);
            return false;
        }
    }
}